src/main/java/io/strimzi/kproxy/api/filter/KrpcFilter.java

```diff
@@ -14,16 +14,12 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.strimzi.kproxy.interceptor;
-
-import io.netty.buffer.ByteBuf;
+package io.strimzi.kproxy.api.filter;
```

Comment:
@tombentley, any objections to getting rid of the "api" name here? There already is an "internal" sub-package elsewhere, i.e. everything which isn't under that one implicitly is API. No need to change it here in this PR, I can do that as a follow-up.

Reply:
That might be a deepish question. I think there _could_ be a case for having an api module (with these interfaces in, and maybe others in due course), a core module (with all the netty bits) and then a number of filter implementation modules. How that maps to packages IDK, so maybe something to think about.

Reply:
Yeah, so what's key is having a proper API/implementation split, i.e. a package and its types should be either one or the other. That's why I had introduced that "internal" package a while ago. How that's mapped to artifacts (API/impl in one module, or multiple ones), is a separate question. Right now things are a bit intermingled again now though: there's two means of expressing something is API (either non-`internal` package, or `api` package. That's what I'd like to rectify before it further proliferates.

Reply:
Pushed one more commit for taking care of this.